art-common/artcommonlib/runtime.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+
+# an abstract class intended to be used by anything looking for config specific to a single
+# ocp-build-data group.
+
+
+class GroupRuntime(ABC):
```

Comment:
If possible, I'd like to not have a common Runtime (otherwise I could imagine that it would grow dramatically). Maybe we can have single-purpose classes and functions for loading group data over parent classes to inherit.

Reply:
i would like to have a common Runtime for things that are actually... common. but not a single consolidated Runtime for everything. even elliott and doozer are different enough to warrant different runtimes, and pyartcd is even more so because it cannot assume group and assembly.

my thought was that `pyartcd.runtime.GroupRuntime` (parallel to the normal runtime), `doozer.runtime`, and `elliott.runtime` would all implement this abstract class, so that when you need to pass in "a runtime" to something related to group and assembly (like the `rhcos` module takes), you know that runtime at least has this class's properties to work with. i didn't yet refactor GroupRuntime into elliott/doozer runtimes but it wouldn't take long...

Reply:
@vfreex there, i put it in 349fed6afd11d6a6e39f9d7f711129fe3b62262e

now, if you really want it to be a separate concept from the other Runtimes, I think it would take a bit of work to untangle it from them, but I can easily just remove that commit and leave it as an adjacent-but-separate concept that elliott and doozer can implement later. what do you think?

Reply:
Your explanation persuaded me. Thanks